metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/common/shell/specials/SpecialCommand.java

```diff
@@ -0,0 +1,60 @@
+/*
+ *
+ *  Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.apache.metron.stellar.common.shell.specials;
+
+import org.apache.metron.stellar.common.shell.StellarShellExecutor;
+import org.apache.metron.stellar.common.shell.StellarResult;
+
+import java.util.function.Function;
+
+/**
+ * A special command that can be run within a Stellar execution
+ * environment.
+ *
+ * Most functionality expected of running Stellar in a shell-like
+ * environment, that is not directly implemented in the Stellar
+ * language itself, is implemented as a SpecialCommand. This
+ * includes magics, doc strings, comments and quit.
+ *
+ * This is typically an action performed on the execution
+ * environment, not something that could be executed within Stellar.
+ */
+public interface SpecialCommand {
```

Comment:
The interface that I used for any logic that we had implemented in the REPL that is not part of the Stellar language.